src/transformers/trainer.py

```diff
@@ -2078,15 +2084,18 @@ def _load_from_checkpoint(self, resume_from_checkpoint, model=None):
         if model is None:
             model = self.model
 
-        if not os.path.isfile(os.path.join(resume_from_checkpoint, WEIGHTS_NAME)) and not os.path.isfile(
-            os.path.join(resume_from_checkpoint, WEIGHTS_INDEX_NAME)
-        ):
+        config_file = os.path.join(resume_from_checkpoint, CONFIG_NAME)
+        weights_file = os.path.join(resume_from_checkpoint, WEIGHTS_NAME)
+        weights_index_file = os.path.join(resume_from_checkpoint, WEIGHTS_INDEX_NAME)
+        safe_weights_file = os.path.join(resume_from_checkpoint, SAFE_WEIGHTS_NAME)
+
+        if not any(map(os.path.isfile, [weights_file, safe_weights_file, weights_index_file])):
```

Comment:
A list comprehension would be faster than map here (it's 4 elements I know, but still)
```suggestion
        if not any([os.path.isfile(f) for f in [weights_file, safe_weights_file, weights_index_file]]):
```